.github/workflows/acquia-backup-databases.yml

```diff
@@ -0,0 +1,99 @@
+  workflow_call:
+    inputs:
+      filter:
+        description: 'Optionally, provide a filter for the databases. See: https://docs.acquia.com/acquia-cli/commands/#api-environments-database-list'
+        type: string
+        required: false
+      php_version:
+        description: 'Optionally, specify a PHP version to install. Default: 8.2.'
+        type: string
+        required: false
+        default: '8.2'
+    secrets:
+      ENVIRONMENT_ID:
+        description: 'Acquia environment ID.'
+        required: true
+      ACLI_CLIENT_ID:
+        description: 'The client ID for authenticating with ACLI.'
+        required: true
+      ACLI_CLIENT_SECRET:
+        description: 'The client secret for authenticating with ACLI.'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  backup-databases:
+    name: 'Backup Acquia databases'
+    runs-on: ubuntu-latest
+    outputs:
+      databases: ${{ steps.get_databases.outputs.database_list }}
+      db_list: ${{ steps.get_databases.outputs.db_list }}
+    steps:
+      - name: Checking out code
+        uses: actions/checkout@v3
+
+      - name: Installing PHP
+        uses: shivammathur/setup-php@master
+        with:
+          php-version: ${{ inputs.php_version }}
+
+      - name: Installing dependencies
+        run: |
+          curl -OL https://github.com/acquia/cli/releases/download/2.13.0/acli.phar
+          echo "ACLI downloaded!"
+
+      - name: Authorizing with ACLI
+        run: |
+          php acli.phar auth:login -n \
+            --key="${{ secrets.ACLI_CLIENT_ID }}" \
+            --secret="${{ secrets.ACLI_CLIENT_SECRET }}"
+
+      - name: Retrieving all database names for this environment
+        id: get_databases
+        # Gets databases filtering by a name. If no filter is provided, it'll
+        # retrieve all databases.
+        run: |
+          # The first time we run api:environments:database-list acli very
+          # kindly also outputs a string asking the user to share
+          # anonymous user data along with the JSON output causing an error
+          # further down the chain - this simply mitigates that.
+          acsf_request=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            --limit='1' \
+            ${{ secrets.ENVIRONMENT_ID }})
+
+          # Gets a list of databases and uses jq to filter the output
+          # to the name value only.
+          database_list=$(php acli.phar api:environments:database-list \
+            --filter='${{ inputs.filter }}' \
+            ${{ secrets.ENVIRONMENT_ID }} \
+            | jq '.[] | .name')
+
+          # Processes the output of databases to allow the data to
+          # be passed to the 'Backing up databases' step.
+          delimiter="$(openssl rand -hex 8)"
+          echo "db_list<<${delimiter}" >> "${GITHUB_OUTPUT}"
+          echo ${database_list} >> ${GITHUB_OUTPUT}
+          echo "${delimiter}" >> "${GITHUB_OUTPUT}"
```

Comment:
Yeah so here, add `-n` (or `--no-interaction`) to simplify this whole section:

```suggestion
          # Gets a list of databases and uses jq to filter the output
          # to the name value only.
          database_list=$(php acli.phar api:environments:database-list \
            --no-interaction
            --filter='${{ inputs.filter }}' \
            ${{ secrets.ENVIRONMENT_ID }} \
            | jq '.[] | .name')

          # Processes the output of databases to allow the data to
          # be passed to the 'Backing up databases' step.
          delimiter="$(openssl rand -hex 8)"
          echo "db_list<<${delimiter}" >> "${GITHUB_OUTPUT}"
          echo ${database_list} >> ${GITHUB_OUTPUT}
          echo "${delimiter}" >> "${GITHUB_OUTPUT}"
```

Question about that second chunk, what does each line do? It looks like you're echoing three different things into the github output - a delimiter, the list, then the delimiter again? 🤔 

Wondering if it could be simplified into one line, or if it could be ommitted entirely. @andrewbelcher might be able to take a look at these and suggest/advise.

Reply:
@iamsophiesk @andrewbelcher  I'm definitely open to suggestions on how we might improve this.

I added this to resolve the following error:
![Screenshot 2023-09-07 at 12 03 35](https://github.com/accessdigital/.github/assets/88322990/4552cb40-e9f3-40dd-bebe-03a95e4e07fc)

The solution implemented was based on the following: https://stackoverflow.com/a/74256214